src/ray/util/process.cc

```diff
@@ -54,7 +55,15 @@ class ProcessFD {
     intptr_t fd;
     pid_t pid;
 #ifdef _WIN32
-    fd = _spawnvp(P_NOWAIT, argv[0], argv);
+    std::vector<std::string> args;
+    for (size_t i = 0; argv[i]; ++i) {
+      args.push_back(argv[i]);
+    }
+    // Calling CreateCommandLine() here wouldn't make sense here if the
+    // Microsoft C runtime properly quoted each command-argument argument.
+    // However, it doesn't quote at all. It just joins arguments with a space.
+    // So we have to do the quoting manually and pass everything as a single argument.
+    fd = _spawnlp(P_NOWAIT, args[0].c_str(), CreateCommandLine(args).c_str(), NULL);
```

Comment:
Calling `.c_str()` on the temporary string returned from `CreateCommandLine(args)` is undefined behaviour (and can blow up), you should store the result in a temporary variable.

Reply:
It's not undefined behavior :) objects are only destroyed at the end of the statement, at which point `_spawnlp` has already returned.

Reply:
looks like it is defined after all: https://stackoverflow.com/a/22146708